### PR TITLE
Chore/pwa integration

### PR DESCRIPTION
**Pull request by chawla-amit**
Features integrated as a part of this Pull Request:

* Added PWA integration for the base next app
* Added basic Manifest file
* Added basic meta tags and manifest linking in document.js
* ~Added static path serving for the custom server for (sw.js and worker-*.js)~
* Added dotenv in the next config for environment variable support
* Added PWA_DISABLED env variable to control whether PWA feature needs to be disabled, by default it will remain enabled.

### CODE REVIEW

File: templates/base/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+PWA_DISABLED=false
```
Comment (vinodloha) [inline templates/base/.env:1]: Not planning to keep any env files this time. Lets avoid having this flag itself. We can consider this as default feature.

File: templates/base/next.config.js [hunk 2]
```diff
@@ -1,13 +1,30 @@
+require('dotenv').config();
```
Comment (vinodloha) [inline templates/base/next.config.js:1]: Lets avoid using this package as there are many ways we can access env variables and .env files is something we are going to discourage as they sometimes lead to security issues.

File: templates/base/next.config.js [hunk 3]
```diff
@@ -1,13 +1,30 @@
...
+/**
+ * This section adds the basic PWA configuration into the next config
+ * It derives from an env variable defined in .env file. Default it is enabled,
+ * To disable it you need to specify the PWA_DISABLED=true in .env file
+ * 
+ * Further PWA configuration visit https://www.npmjs.com/package/next-pwa
+ */
+const nextConfig = withPWA({
+  pwa: {
+    disable: process.env.PWA_DISABLED === 'true',
```
Comment (vinodloha) [inline templates/base/next.config.js:18]: Feel Free to use it this way or define this in docker file.

File: templates/base/next.config.js [hunk 4]
```diff
@@ -1,13 +1,30 @@
...
 const isAssetPrefix = process.env.BASE_PATH || '';
 
+/**
+ * This section adds the basic PWA configuration into the next config
+ * It derives from an env variable defined in .env file. Default it is enabled,
+ * To disable it you need to specify the PWA_DISABLED=true in .env file
+ * 
+ * Further PWA configuration visit https://www.npmjs.com/package/next-pwa
+ */
+const nextConfig = withPWA({
```
Comment (vinodloha) [inline templates/base/next.config.js:16]: For the sake of maintainability can we not maintain this PWA config under /base/config/pwa/config. ? That can export nextConfig and we can apply it here.

File: templates/base/next.config.js (hunk 3 above)
Comment (chawla-amit) [reply templates/base/next.config.js:18]: (y)

File: templates/base/next.config.js (hunk 4 above)
Comment (chawla-amit) [reply templates/base/next.config.js:16]: I wasn't aware of , let me check

File: templates/base/next.config.js (hunk 2 above)
Comment (chawla-amit) [reply templates/base/next.config.js:1]: sure will remove it.

File: templates/base/.env (hunk 1 above)
Comment (chawla-amit) [reply templates/base/.env:1]: will remove it.

**Comment by chawla-amit** [comment]
@vinodloha  Review Comments incorporated. Please have a look.

File: templates/base/src/pages/_document.js [hunk 5]
```diff
@@ -10,7 +10,19 @@ class MyDocument extends Document {
   render() {
     return (
       <Html lang="en">
-        <Head />
+        <Head>
+          <meta name='application-name' content='Universal React V2' />
```
Comment (vinodloha) [inline templates/base/src/pages/_document.js:14]: This should be done via a component, so that conditionally we are able to add or skip it.

**Pull request closed without merging**